lib/sink.js

```diff
@@ -8,52 +8,57 @@ const path = require('path');
 const fs = require('fs');
 
 
-const SinkFs = module.exports = function (fileDir) {
-    if (!(this instanceof SinkFs)) { return new SinkFs(fileDir); }
-    assert(fileDir, '"fileDir" must be provided');
+module.exports = class SinkFs {
+    constructor (fileDir) {
+        assert(fileDir, '"fileDir" is missing');
```

Comment:
Lets do this in a separate task. There is more to this:

 - Temp file should be written to the temp directory of the OS. Not the same directory as the destination of the file.
 - There should be a check which throws if destination directory does not exist.
 - I also think the constructor of the sinks should take a options object instead of an argument like this. This sink is pretty simple, but the config one need to pass in to ex a S3 sink is more complex.